### PR TITLE
ddclient: update source & homepage

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
follow up from #125232 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Stunkymonkey** [comment]
@SuperSandro2000 any additional things, i missed out?

File: pkgs/tools/networking/ddclient/default.nix [hunk 1]
```diff
@@ -5,7 +5,7 @@ perlPackages.buildPerlPackage rec {
   version = "3.9.1";
 
   src = fetchurl {
-    url = "mirror://sourceforge/ddclient/${pname}-${version}.tar.gz";
+    url = "https://github.com/ddclient/${pname}/archive/refs/tags/v${version}.tar.gz";
```
Comment (ajs124) [inline pkgs/tools/networking/ddclient/default.nix:8]: This should probably be using `fetchFromGitHub`

Comment (SuperSandro2000) [reply pkgs/tools/networking/ddclient/default.nix:8]: Yes, it should.

File: pkgs/tools/networking/ddclient/default.nix [hunk 2]
```diff
@@ -38,9 +38,9 @@ perlPackages.buildPerlPackage rec {
 
   meta = with lib; {
     description = "Client for updating dynamic DNS service entries";
-    homepage    = "https://sourceforge.net/p/ddclient/wiki/Home/";
-    license     = licenses.gpl2Plus;
+    homepage = "https://ddclient.net/";
+    license = licenses.gpl2Plus;
     # Mostly since `iproute` is Linux only.
-    platforms   = platforms.linux;
+    platforms = platforms.linux;
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/ddclient/default.nix:44]:
```suggestion
    platforms = platforms.linux;
    maintainers = with maintainers; [ SuperSandro2000 ];
```

I am one of the upstream devs but currently a bit inactive. Nevertheless.

**Review by SuperSandro2000 — COMMENTED** [review]
thanks!

**Pull request merged**